### DISCUSSION TITLE
how to define a discussion/reviewing process

### DISCUSSION THREAD

**Discussion by ft2023** [ulab-uiuc/research-town#55] [Q&A]
1.We should assign an order for all the discussion/review agents, since the cooperation process requires the later agents to get the information of  previous agents
2. Based on the information of other agents to construct the communication mechanism of multiple agents
3. Also we can define a meta agent to summarize the info of all the agents

plus: maybe we can check the paper "AgentVerse: Facilitating Multi-Agent Collaboration and Exploring Emergent Behaviors in Agents" for more details of multi-agents discussion/review

**Comment by chengzr01** [comment]
I think the ideas in [AgentVerse](https://arxiv.org/abs/2308.10848) can be very useful for us. There are **separated meta-agents** for expert recruitment (e.g., role assigners) and **various structures** for collective decision-making (e.g., horizontal structure and vertical structure) in their settings. I am generally in favor of integrating such components or implementing such techniques into our framework. For example, when reviewing papers, we also need agents that can assign papers to reviewers and mechanisms that can gather decisions from reviewers.
<img width="879" alt="image" src="https://github.com/ulab-uiuc/research-town/assets/112863445/578585f8-1df9-4325-a4e8-faa5bfe696c0">

**Reply by chengzr01** [reply]
However, I think we might have different motivations than AgentVerse. They claim that their contribution is to propose a framework that effectively orchestrates a collaborative group of expert agents, implying that the meta-agents are actually exogenous to the environment. Therefore, I wonder whether we need to implement some endogenous mechanisms within the environment? For example, when reviewing papers, should we let the agents elect their own reviewers instead of having a meta-agent assign the reviewers?

**Reply by lwaekfjlk** [reply]
@ft2023 what do you think of this?

@chengzr01 Thanks for mentioning this. I think it is super good and valuable points. What do you think we should do to integrate such functions into our current framework. What kind of architecture change should we make? For example, is meta-agent a special class or sub-class of agent? Should collaboration be defined as an env class or functions in agent? Maybe a potential class map can be useful and helpful for guidance?

**Reply by chengzr01** [reply]
Thank you for your questions! I am actually looking forward to valuable opinions from all of you! Here are some of my initial thoughts:  I agree that task-related things such as collaboration and discussion should be defined as properties of environments. I think we may not necessarily need to implement a special class of meta-agent as in AgentVerse. A potential class map would be useful in our context. Maybe we can also consider designing subclasses inheriting from base agent classes to play extra roles later.

**Reply by lwaekfjlk** [reply]
Thanks @chengzr01, makes sense. We can discuss this in more details when facing more complicated tasks.

**Reply by chengzr01** [reply]
Sounds great!

**Comment by ft2023** [comment]
great! I think for the reviewer assignment, we need a retrieval-based function to retrieve suitable reviewers based on the topics of the papers and the profiles of reviewers. indeed, AgentVerse can bring us a lot of useful advice on how to design good prompts during the reviewing process.

**Reply by lwaekfjlk** [reply]
based on my understanding, all retrieva-based function should be included in kb? What do you think

**Comment by lwaekfjlk** [comment]
Conceptually, I feel like all the things related to a lot of things like reviewer db / paper db should all be implemented in kb class (might have separate sub-class)?

All the task-related things like discussion / review / submit should be defined as an envioronment.
